src/auto4_lua.cpp

```diff
@@ -270,6 +271,20 @@ namespace {
 		return 2;
 	}
 
+	int lua_set_status_text(lua_State *L)
+	{
+		const agi::Context *c = get_context(L);
+		if (!c || !c->frame) {
+			lua_pushnil(L);
+			return 1;
+		}
+		std::string text = check_string(L, 1);
+		lua_pop(L, 1);
+		wxString wx_text(text.c_str(), wxConvUTF8);
```

Comment:
`agi::dispatch::Main().Async([=] { c->frame->SetStatusText(to_wx(text)); }`

Lua scripts are run on a background thread and GUI stuff can only be touched on the main thread. This also probably actually wants to be calling `StatusTimeout()` rather than setting a persistent thing.